File: 2st/Riyeon/유한소수 판별하기.py
import math
def solution(a, b):
    b = b / math.gcd(a, b)  # 최대 공약수
    
    while b % 2 == 0:
        b /= 2
    while b % 5 == 0:
        b /= 5
    
    return 1 if b == 1 else 2

File: 4st/Eunseong/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        for (int i = a; i >= 1; i--) {
            if (a % i == 0) {
                if (b % i == 0) {
                    a /= i;
                    b /= i;
                    break;
                }
            }
        }
        while (b != 1) {
            if (b % 2 == 0) b /= 2;
            else if (b % 5 == 0) b /= 5;
            else return 2;
        }
        return 1;
    }
}